File: src/frontend/src/app/core/models/catalog.models.ts
export interface ProductImage {
  id: string;
  url: string;
  altText?: string | null;
  displayOrder: number;
  isPrimary: boolean;
}

export interface CategorySummary {
  id: string;
  name: string;
  slug: string;
}

export interface ProductListItem {
  id: string;
  name: string;
  slug: string;
  sku: string;
  shortDescription?: string | null;
  price: number;
  promotionalPrice?: number | null;
  stockQuantity: number;
  isActive: boolean;
  category: CategorySummary;
  primaryImage?: ProductImage | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface ProductDetail extends ProductListItem {
  longDescription?: string | null;
  images: ProductImage[];
}

export interface PagedResult<T> {
  items: T[];
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

export interface ProductFilters {
  page?: number;
  pageSize?: number;
  categoryId?: string;
  minPrice?: number;
  maxPrice?: number;
  isActive?: boolean;
  sortBy?: 'name' | 'price' | 'createdAt';
  sortDirection?: 'asc' | 'desc';
  search?: string;
}

File: src/backend/src/Modules/Catalog/Ecommerce.Modules.Catalog.Application/Models/CatalogModels.cs
namespace Ecommerce.Modules.Catalog.Application.Models;

public sealed record PagedResult<T>(
    IReadOnlyList<T> Items,
    int Page,
    int PageSize,
    int TotalCount,
    int TotalPages);

public sealed record ProductImageDto(
    Guid Id,
    string Url,
    string? AltText,
    int DisplayOrder,
    bool IsPrimary);

public sealed record CategorySummaryDto(
    Guid Id,
    string Name,
    string Slug);

public sealed record CategoryResponse(
    Guid Id,
    string Name,
    string Slug,
    string? Description,
    Guid? ParentCategoryId,
    bool IsActive,
    DateTime CreatedAt,
    DateTime? UpdatedAt);

public sealed record ProductListItemResponse(
    Guid Id,
    string Name,
    string Slug,
    string Sku,
    string? ShortDescription,
    decimal Price,
    decimal? PromotionalPrice,
    int StockQuantity,
    bool IsActive,
    CategorySummaryDto Category,
    ProductImageDto? PrimaryImage,
    DateTime CreatedAt,
    DateTime? UpdatedAt);

public sealed record ProductDetailResponse(
    Guid Id,
    string Name,
    string Slug,
    string Sku,
    string? ShortDescription,
    string? LongDescription,
    decimal Price,
    decimal? PromotionalPrice,
    int StockQuantity,
    bool IsActive,
    CategorySummaryDto Category,
    IReadOnlyList<ProductImageDto> Images,
    DateTime CreatedAt,
    DateTime? UpdatedAt);

public sealed record ProductImageInput(
    string Url,
    string? AltText,
    int DisplayOrder,
    bool IsPrimary);
